lib/operations/insert_one.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+const applyRetryableWrites = require('../utils').applyRetryableWrites;
+const applyWriteConcern = require('../utils').applyWriteConcern;
+const handleCallback = require('../utils').handleCallback;
+const MongoError = require('mongodb-core').MongoError;
+const OperationBase = require('./operation').OperationBase;
+const toError = require('../utils').toError;
+
+class InsertOneOperation extends OperationBase {
+  constructor(collection, doc, options) {
+    super(options);
+
+    this.collection = collection;
+    this.doc = doc;
+  }
+
+  execute(callback) {
+    const coll = this.collection;
+    const doc = this.doc;
+    const options = this.options;
+
+    if (Array.isArray(doc)) {
+      return callback(
+        MongoError.create({ message: 'doc parameter must be an object', driver: true })
+      );
+    }
+
+    insertDocuments(coll, [doc], options, (err, r) => {
```

Comment:
Note for future: Once we have more operations defined, I'm interested in how these common behaviors are shared between them.